### DISCUSSION TITLE
How can I configure vercel.json to rewrite requests with empty path

### DISCUSSION THREAD

**Discussion by ymc9** [vercel/community#1141] [Help]
Hi guys,

I'm using vercel.json's `rewrites` to serve SEO friendly content to search engine crawlers. I can't figure out what to use for "source" field if I want to rewrite request with empty path, i.e., https://mydomain.com -> https://mydomain.com/static/index.html.

I tried using "/", "", and "$^" for "source" field but none seems to work. Ideas are greatly appreciated. Thanks!

My vercel.json:
```json
{
    "trailingSlash": false,
    "rewrites": [
        {
            "source": "",
            "has": [
                {
                    "type": "header",
                    "key": "user-agent",
                    "value": "Googlebot.*"
                }
            ],
            "destination": "/static/index.html"
        },
        {
            "source": "/:path*",
            "has": [
                {
                    "type": "header",
                    "key": "user-agent",
                    "value": "Googlebot.*"
                }
            ],
            "destination": "/static/:path*.html"
        },
        { "source": "/sitemap.xml", "destination": "/sitemap.xml" },
        { "source": "/:path*", "destination": "/index.html" }
    ]
}

```

**Comment by amyegan** [comment]
Thanks for reaching out @ymc9! If you only want to match the root path, the `source` value should be `"/"`. 

Since you said you already tried that, I'm going to need to look into this further to figure out what is going wrong. Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) or public repo that I can use to see the problem and debug?

**Reply by ymc9** [reply]
Hi @amyegan ,

Thanks for following up. I've got a repro project here: https://github.com/ymc9/vercel-rewrite-repro, deployed at: https://vercel-rewrite-repro.vercel.app.

As you can see, path "/hello" gets rewritten correctly, but root path (with or without trailing "/") doesn't work.

**Reply by amyegan** [reply]
This repro is very helpful ❤️ 

The team is taking a look, but I don't have an ETA on a solution. As a workaround, you can rewrite using the Edge Middleware instead. There is a [quickstart guide](https://vercel.com/docs/concepts/functions/edge-middleware/quickstart), a [template](https://vercel.com/templates/next.js/hostname-rewrites), and an [example repo](https://github.com/vercel/examples/blob/main/edge-functions/hostname-rewrites/middleware.ts) available to help with setup.

**Reply by smaeda-ks** [reply]
Hi @ymc9, from the example repo above, I believe it will work if you delete the `index.html` file. Can you confirm?

If you need to rewrite on `/`, you probably need to delete the corresponding page from your repository. Otherwise, it takes precedence over.
